02-express-tutorial/app.js

```diff
@@ -1 +1,72 @@
-console.log('Express Tutorial')
+console.log("Express Tutorial");
+
+const express = require("express");
+const { products } = require("./data");
+
+const app = express(); //  npm start -> http://localhost:3000
+app.use(express.static("./public"));
+
+app.get("/api/v1/test", (req, res) => { // http://localhost:3000/api/v1/test
+    res.json({ msg: "It worked!" });
+});
+
+app.get("/api/v1/products", (req, res) => { // http://localhost:3000/api/v1/products
+    res.json(products);
+});
+
+app.get("/api/v1/products/:productID", (req, res) => {
+    const idToFind = parseInt(req.params.productID);
+    const product = products.find((p) => p.id === idToFind);
+    if (product) {
+        res.json(product); // http://localhost:3000/api/v1/products/1 (valid ID)
+    } else {
+        res.status(404).json({ message: "Product was not found." }); // http://localhost:3000/api/v1/products/1000 (invalid ID)
+    }
+});
+
+/*
+app.get("/api/v1/query", (req, res) => {
```

Comment:
Now that you are familiar with the use of git, You can remove commented code blocks like this!

Reply:
Should you need this again, you can retrieve it from your git history.